### PR TITLE
Split gather_network_logs into basics and extras

### PR DESCRIPTION
**Pull request by flavio-fernandes**
With OVNK-Interconnect support in 4.14, ovnkube-nodes logs grew large. This commit adds the ability to either collect or skip the additional logs.

Script changes:
- collection-scripts/gather_network_logs_basics now used as the "regular" gather script.
- Renamed collection-scripts/gather_network_logs to collection-scripts/gather_network_logs_basics.
- Improved gather_network_logs to invoke gather and collect additional logs beyond basics.

Reference: https://issues.redhat.com/browse/SDN-3733
Reported-at: https://issues.redhat.com/browse/SDN-3507

### CODE REVIEW

**Comment by flavio-fernandes** [comment]
This PR should be merged after https://github.com/openshift/cluster-network-operator/pull/1938

File: collection-scripts/gather_network_logs_basics [hunk 1]
```diff
@@ -0,0 +1,148 @@
+#!/bin/bash
```
Comment (flavio-fernandes) [inline collection-scripts/gather_network_logs_basics:1]: Note: file is actually, the original `gather_network_logs` file.

File: collection-scripts/gather_network_logs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+# start off by doing a gather, which includes gather_network_logs_basics.
+/usr/bin/gather
+
+BASE_COLLECTION_PATH="must-gather"
+NETWORK_LOG_PATH=${OUT:-"${BASE_COLLECTION_PATH}/network_logs"}
+OVNK_EXTRAS_STORE_PATH=${OUT:-"ovnk_extras_store"}
+
+mkdir -p "${NETWORK_LOG_PATH}"/
+
+function gather_extra_ovn_kubernetes_data_interconnect_mode {
```
Comment (flavio-fernandes) [inline collection-scripts/gather_network_logs:12]: .

File: collection-scripts/gather_network_logs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  sample_node_zone=$(oc get node "${sample_node}" -o jsonpath='{.metadata.annotations.k8s\.ovn\.org/zone-name}')
+  if [ "${sample_node}" = "${sample_node_zone}" ]; then
+    gather_extra_ovn_kubernetes_data_interconnect_mode
+  fi
+}
+
+PIDS=()
+NETWORK_TYPE=$(oc get network.config.openshift.io -o=jsonpath='{.items[0].spec.networkType}' | tr '[:upper:]' '[:lower:]')
+
+if [[ "${NETWORK_TYPE}" == "kuryr" ]]; then
```
Comment (npinaeva) [inline collection-scripts/gather_network_logs:39]: maybe we can just exit here if `NETWORK_TYPE` != `ovnkubernetes`?

File: collection-scripts/gather_network_logs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      oc rsync --include='*log*' --exclude='*' -n openshift-ovn-kubernetes -c ${CONTAINER} \
+        ${OVNKUBE_CONTROLLER_POD}:/var/log/ovnkube/ ${DEST_DIR} 2>&1 & PIDS+=($!)
+    done
+  done
+}
+
+function gather_extra_ovn_kubernetes_data {
+  sample_node=$(oc get no -o jsonpath='{.items[0].metadata.name}')
+  sample_node_zone=$(oc get node "${sample_node}" -o jsonpath='{.metadata.annotations.k8s\.ovn\.org/zone-name}')
+  if [ "${sample_node}" = "${sample_node_zone}" ]; then
```
Comment (npinaeva) [inline collection-scripts/gather_network_logs:31]: in which case we expect this condition to be false?

File: collection-scripts/gather_network_logs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+OVNK_EXTRAS_STORE_PATH=${OUT:-"ovnk_extras_store"}
+
+mkdir -p "${NETWORK_LOG_PATH}"/
+
+function gather_extra_ovn_kubernetes_data_interconnect_mode {
+  echo "INFO: Gathering ovn-kubernetes extra logs"
+  OVNKUBE_CONTROLLER_PODS=($(oc -n openshift-ovn-kubernetes get pods -l app=ovnkube-node -o=jsonpath='{.items[*].metadata.name}'))
+  # collect additional ovnkube logs from each node
+  CONTAINERS="ovnkube-controller"
+  for CONTAINER in "${CONTAINERS}" ; do
```
Comment (npinaeva) [inline collection-scripts/gather_network_logs:17]: do we need this loop?

**Review by npinaeva — COMMENTED** [review]
some nits about the script, but also @flavio-fernandes can you please share a must-gather archive that you get by running this new command so that I don't need to spin up a cluster just for that?

**Review by soltysh — APPROVED** [review]
/approve
the general idea
@npinaeva has final lgtm on this PR

File: collection-scripts/gather_network_logs [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    for OVNKUBE_CONTROLLER_POD in "${OVNKUBE_CONTROLLER_PODS[@]}"; do
+      echo "INFO: Gathering "${CONTAINER}" extra logs from "${OVNKUBE_CONTROLLER_POD}"..."
+      DEST_DIR="${NETWORK_LOG_PATH}/${OVNK_EXTRAS_STORE_PATH}/${OVNKUBE_CONTROLLER_POD}/${CONTAINER}"
+      mkdir -p "${DEST_DIR}"
+      oc rsync --include='*log*' --exclude='*' -n openshift-ovn-kubernetes -c ${CONTAINER} \
```
Comment (pperiyasamy) [inline collection-scripts/gather_network_logs:22]: @flavio-fernandes can we still narrow down to exact log file matching pattern ? does this `libovsdb.log*` work ?

File: collection-scripts/gather_network_logs (hunk 5 above)
Comment (flavio-fernandes) [reply collection-scripts/gather_network_logs:17]: heh, we do not at all. I was trying to think ahead if we would ever care for other containers in that pod.
But less is more; we can add it should the need arise.

File: collection-scripts/gather_network_logs (hunk 3 above)
Comment (flavio-fernandes) [reply collection-scripts/gather_network_logs:39]: sure thing. Let me simplify  that.

File: collection-scripts/gather_network_logs (hunk 6 above)
Comment (flavio-fernandes) [reply collection-scripts/gather_network_logs:22]: I think so. I'd suspect that any logs in here should be part of the mg, but let me make it a bit more specific.

File: collection-scripts/gather_network_logs (hunk 4 above)
Comment (npinaeva) [reply collection-scripts/gather_network_logs:30]: backwards compatibility makes perfect sense! I am just trying to figure out if we only need to collect extra logs in 1-node-per-zone config (e.g. is there a possible default zone or some other type of config where we would have libovsdb logs in another file)?

Comment (flavio-fernandes) [reply collection-scripts/gather_network_logs:37]: ah. ic. no, iirc there is no space issues when we only have 3 ovnkube-controllers, so nothing additional here for non-ovn-ic cases.

File: collection-scripts/gather_network_logs [hunk 7]
```diff
@@ -1,148 +1,53 @@
...
-CNCC_DEPLOYMENT=$(oc get deployment -n openshift-cloud-network-config-controller --no-headers -o custom-columns=":metadata.name")
-if [[ "${CNCC_DEPLOYMENT}" == "cloud-network-config-controller" ]]; then
-  oc adm inspect --dest-dir must-gather cloudprivateipconfigs.cloud.network.openshift.io
+if [[ "${NETWORK_TYPE}" != "ovnkubernetes" ]]; then
+    exit 0
 fi
 
-echo "INFO: Waiting for network log collection to complete ..."
+echo "INFO: Waiting for network extra logs collection to complete ..."
 if [[ "${NETWORK_TYPE}" == "ovnkubernetes" ]]; then
```
[the remark this answers is not in the document]
Comment (flavio-fernandes) [reply collection-scripts/gather_network_logs:51]: true! ;)

**Comment by flavio-fernandes** [comment]
@npinaeva I uploaded the MG here:  https://drive.google.com/drive/folders/1eXyp_wJ5-VoiDKaZv8xg4cMgFOHM_oH4?usp=drive_link

In the process, I hit an error I did not expect when using the mg images: no rsync! :( 
so I'm super glad you asked mt to try it because I was assuming it would work just as well outside the mg image.
The drive with link above includes the commands I tried:
```
❯ oc adm must-gather --image=quay.io/ffernand/mg:${TAG} -- gather_network_logs 2>&1 | tee ~/collect.txt
❯ tar tzvf ./must-gather.loc*/quay-io-ffernan*/network_logs/ovnk_extras_store.tar.gz
```

**Comment by npinaeva** [comment]
/lgtm
thanks @flavio-fernandes !

**Pull request merged**